cli/tests/test_next_prev_commands.rs

```diff
@@ -567,6 +567,126 @@ fn test_next_editing() {
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["prev", "--conflict", "--edit"]);
+    // We now should be editing the third commit.
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: kkmpptxz 26b1439f (conflict) third
+    Parent commit      : rlvkpnrz 55b5d11a (empty) second
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+    "###);
+}
+
+#[test]
+fn test_next_conflict() {
+    // There is a conflict in the third commit, so after next it should be the new
+    // parent.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    std::fs::write(&file_path, "second").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    // Create a conflict in the third commit.
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(first)"]);
+    std::fs::write(&file_path, "first").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(second)"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(first)"]);
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["next", "--conflict"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: vruxwmqv b69eca51 (conflict) (empty) (no description set)
+    Parent commit      : rlvkpnrz fa43d820 (conflict) second
+    Added 0 files, modified 1 files, removed 0 files
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+    "###);
+    // --edit is implied when already editing a non-head commit
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["next"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: yostqsxw e4ce14ca (conflict) (empty) (no description set)
+    Parent commit      : mzvwutvl d37ca916 (conflict) (empty) third
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+    "###);
+}
+
+#[test]
+fn test_next_conflict_editing() {
+    // There is a conflict in the third commit, so after next it should be our
+    // working copy.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    std::fs::write(&file_path, "second").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    // Create a conflict in the third commit.
+    std::fs::write(&file_path, "first+1").unwrap();
```

Comment:
`first+1` is confusing content for something that goes on top of the content `second`. Just use `third` (or `second+1` if you prefer)?

Reply:
done